scripts/uploader.rb

```diff
@@ -22,10 +22,9 @@ def initialize(directory, type)
     def run
       login
 
-      appliances = Dir.glob("#{directory}/manageiq*")
-      appliances.each do |appliance|
-        # Skip badly named appliances, missing git sha: manageiq-openstack-master-201407142000-.qc2
-        unless appliance.match?(/.+-[0-9]{12}-[0-9a-fA-F]+/)
+      Dir.glob("#{directory}/*").each do |appliance|
+        # Skip files without date
+        unless appliance.match?(/.+-[0-9]{12}/)
```

Comment:
Why are we not checking for the git sha like previously?

Reply:
Because v2v appliance won't have SHA in its name, so the only thing guaranteed to be in all images (but not included in non-image files) is the date/time.

If we're to include SHA in the v2v image as well, that will be the SHA of the repo where kickstart file will come from. But the kickstart file will rarely change once we finalize it, so it's kind of misleading to have SHA. The main contents of the image come from v2v-conversion-host container image (it's pulled inside appliance).

Reply:
The previous check though was to prevent the build name not having a SHA in it

    # Skip badly named appliances, missing git sha: manageiq-openstack-master-201407142000-.qc2

so the new regex will still allow that "badly named appliance".

> But the kickstart file will rarely change once we finalize it, so it's kind of misleading to have SHA.

I guess it depends what the kickstart file is doing... does it pull in some source?  If so, then a SHA (of the code in general, not of just the file) would make sense.

Reply:
> so the new regex will still allow that "badly named appliance".

I've never seen images created without SHA (or any bad names for that matter)... so to me that check was more for filtering out non-image files that exist in that directory, than really filtering out bad named appliances.

I'd hope something will fail earlier if we don't even have the manageiq core repo SHA!

> I guess it depends what the kickstart file is doing... does it pull in some source? If so, then a SHA (of the code in general, not of just the file) would make sense.

The appliance build will pull in container image. The container image (Dockerfile is in the same [repo](https://github.com/ManageIQ/manageiq-v2v-conversion_host-build) as kickstart file - which probably won't change too often, but more often than kickstart file) pulls bunch of RPMs. Probably we can consider v2v-conversion-host-wrapper RPM as the "source"(?) which is in yet another [repo](https://github.com/ManageIQ/manageiq-v2v-conversion_host). But since that's a released RPM, we have to find out the latest version and get the SHA of the repo if we want to go that route.